blocks/donation-form/class-give-donation-form-block.php

```diff
@@ -21,200 +21,219 @@
+     * Class Constructor
+     *
+     * Set up the Give Donation Grid Block class.
+     *
+     * @since  2.0.2
+     * @access private
+     */
+    private function init()
+    {
+        add_action('init', array($this, 'register_block'), 999);
+        add_action('wp_ajax_give_block_donation_form_search_results', array($this, 'block_donation_form_search_results')
+        );
+        add_filter('rest_prepare_give_forms', array($this, 'addExtraDataToResponse'), 10, 2);
+    }
+
+    /**
+     * Add extra data to response.
+     *
+     * @since 2.7.0
+     * @param WP_REST_Response $response
+     * @param WP_Post $form
+     *
+     * @return WP_REST_Response
+     */
+    public function addExtraDataToResponse($response, $form)
+    {
+        // Return extra data only if query in edit context.
+        if (empty($_REQUEST['context']) || $_REQUEST['context'] !== 'edit') {
+            return $response;
+        }
+
+        $data = $response->get_data();
+        $data['formTemplate'] = Give()->form_meta->get_meta($form->ID, '_give_form_template', true);
+        $response->set_data($data);
+
+        return $response;
+    }
+
+    /**
+     * Register block
+     *
+     * @since  2.1
+     * @access public
+     *
+     * @access public
+     */
+    public function register_block()
+    {
+        // Bailout.
+        if (!function_exists('register_block_type')) {
+            return;
+        }
+
+        // Register block.
+        register_block_type(
+            'give/donation-form',
+            array(
+                'render_callback' => array($this, 'render_donation_form'),
+                'attributes' => array(
+                    'id' => array(
+                        'type' => 'number',
+                    ),
+                    'prevId' => array(
+                        'type' => 'number',
+                    ),
+                    'displayStyle' => array(
+                        'type' => 'string',
+                        'default' => 'onpage',
+                    ),
+                    'continueButtonTitle' => array(
+                        'type' => 'string',
+                        'default' => '',
+                    ),
+                    'showTitle' => array(
+                        'type' => 'boolean',
+                        'default' => true,
+                    ),
+                    'showGoal' => array(
+                        'type' => 'boolean',
+                        'default' => true,
+                    ),
+                    'contentDisplay' => array(
+                        'type' => 'boolean',
+                        'default' => true,
+                    ),
+                    'showContent' => array(
+                        'type' => 'string',
+                        'default' => 'above',
+                    ),
+                    'blockId' => array(
+                        'type' => 'string',
+                    ),
+                    'openFormButton' => array(
```

Comment:
Also, `openFormButton` is the same thing as `continueButtonTitle`. The same goes for `formFormat` which is `displayStyle`. We don't actually need new attributes, so let's use what we already have for backward compatibility.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7082/commits/f93b8bd8e2d19e874fe0c32586a726dcf9704909